lib/msf/core/post/file.rb

```diff
@@ -73,8 +73,14 @@ def dir(directory)
       return session.fs.dir.entries(directory)
     end
 
+    if session.type == 'powershell'
+      dir = session.shell_command_token("Get-ChildItem \"#{directory}\" | Format-Table Name").split(/[\r\n]+/)
+      dir.slice!(0..2) if dir.length > 2
+      return dir
+    end
+
     if session.platform == 'windows'
-      return session.shell_command_token("dir #{directory}").split(/[\r\n]+/)
+      return session.shell_command_token("dir /b \"#{directory}\"").split(/[\r\n]+/)
```

Comment:
You probably want to protect this split from nil:
`/home/tmoose/rapid7/metasploit-framework/lib/msf/core/post/file.rb:77:in `dir': undefined method `split' for nil:NilClass (NoMethodError)`

Reply:
Done :)